src/MySqlConnector/MySqlClient/ConnectionPool.cs

```diff
@@ -241,5 +280,7 @@ private ConnectionPool(ConnectionSettings cs)
 		readonly SemaphoreSlim m_sessionSemaphore;
 		readonly LinkedList<MySqlSession> m_sessions;
 		readonly ConnectionSettings m_connectionSettings;
+		readonly Dictionary<int, WeakReference<MySqlSession>> m_leasedSessions;
+		int m_lastId;
```

Comment:
`long` or `ulong` may be better. Although it'd be rare, `int`could overflow in a high load scenario given enough time